core/src/main/java/software/amazon/awssdk/async/AsyncResponseHandler.java

```diff
@@ -46,6 +49,8 @@
      * will be treated as a failure of the response and the {@link #exceptionOccurred(Throwable)} callback will be invoked.
      * </p>
      *
+     * <p>This callback may never be called if the response has no content or if an error occurs.</p>
```

Comment:
No from the perspective of this interface it means any content. For non streaming content we implement a response handler that dumps it all into memory and calls the sync unmarshaller.